unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AudioSource/ECSAudioSourceComponentHandler.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Helpers;
+using DCL.Models;
+using DCL.SettingsCommon;
+using UnityEngine;
+using AudioSettings = DCL.SettingsCommon.AudioSettings;
+
+namespace DCL.ECSComponents
+{
+    public class ECSAudioSourceComponentHandler : IECSComponentHandler<ECSAudioSource>, IOutOfSceneBoundariesHandler
+    {
+        internal AudioSource audioSource;
+        internal AssetPromise_AudioClip promiseAudioClip;
+        
+        private long playedAtTimestamp = 0;
+        private bool isOutOfBoundaries = false;
+        
+        private ECSAudioSource model;
+        private IParcelScene scene;
+        private AudioClip audioClip;
+
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
+        {
+            audioSource = entity.gameObject.AddComponent<AudioSource>();
+
+            if (Settings.i != null)
+                Settings.i.audioSettings.OnChanged += OnAudioSettingsChanged;
+    
+            DataStore.i.sceneBoundariesChecker.Add(entity,this);
+            DataStore.i.virtualAudioMixer.sceneSFXVolume.OnChange += OnVirtualAudioMixerChangedValue;
+        }
+
+        public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+        {
+            Dispose();
+            DataStore.i.sceneBoundariesChecker.Remove(entity,this);
```

Comment:
could we put this remove inside the `Dispose`? looks kind of weird having all unsubscriptions done inside `Dispose` and this one out

Reply:
Yeah we can move it and make the dispose coupled with the entity